testplan/common/utils/convert.py

```diff
@@ -233,12 +240,26 @@ def flatten_dict_comparison(comparison):
     while left or right:
         lpart, rpart = None, None
         if left and right:
-            if len(left[0][0]) > len(right[0][0]):
+            # NOTE: if the left keypath is longer we entered a nested structure
+            #           on one side only
+            #           if the key is Absent only on left side, then we just insert an
+            #           empty row for visual separation
+            if (
+                len(left[0][0]) > len(right[0][0])
+                or left[0][2] is Absent
+                and left[0][2] != right[0][2]
+            ):
                 lpart = left.pop(0)
-            elif len(left[0][0]) < len(right[0][0]):
+            # NOTE: same as above but from right-hand side perspective
+            elif (
+                len(left[0][0]) < len(right[0][0])
+                or right[0][2] is Absent
+                and left[0][2] != right[0][2]
+            ):
```

Comment:
These two updates to the conditionals are the actual changes. The row that is causing the mismatch upon the scenario discussed is readily identified (key is Absent on exactly one side). The behavior should be as in case of longer keypath so it is just incremental condition to the keypath length check, then handling should be identical.